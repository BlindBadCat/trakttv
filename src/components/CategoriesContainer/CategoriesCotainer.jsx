import React from 'react';
import PropTypes from 'prop-types';

const CategoriesContainer = ({ handleSelect }) => (
  <div className="col-sm">
    <div style={{ fontSize: 30 }} className="col-sm text-center">Genre:</div>
    <select className="custom-select sm" onChange={handleSelect}>
      <option value="">None</option>
      <option value="action">Action</option>
      <option value="anime">Anime</option>
      <option value="drama">Drama</option>
      <option value="fantasy">Fantasy</option>
      <option value="horror">Horror</option>
      <option value="musical">Musical</option>
      <option value="sports">Sports</option>
      <option value="thriller">Thriller</option>
      <option value="war">War</option>
    </select>
  </div>
);

CategoriesContainer.propTypes = {
  handleSelect: PropTypes.func.isRequired,
};

export default CategoriesContainer;
