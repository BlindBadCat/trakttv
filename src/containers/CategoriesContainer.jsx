import React from 'react';
import { NavLink } from 'react-router-dom';
import { withRouter } from 'react-router';
import HeadNavigationLinksComponent from '../components/HeadNavigationLinksComponent';
import C from '../constants';

class CategoriesContainer extends React.Component {
  render() {
    const { match } = this.props;
    const { params } = match;
    const {
      sort = 'watched', query = '',
    } = params;

    return (
      <HeadNavigationLinksComponent title="Genre">
        {C.GENRES.map(genre => (
          <NavLink
            to={`/${sort}/${genre}/1${query ? `/${query}` : ''}`}
            style={{ flex: 1, listStyleType: 'none' }}
            activeClassName="selected"
            className="btn btn-dark"
          >
            {`${genre.charAt(0).toUpperCase()+genre.slice(1)}`}
          </NavLink>
        ))}
      </HeadNavigationLinksComponent>
    );
  }
}


export default withRouter(CategoriesContainer);
