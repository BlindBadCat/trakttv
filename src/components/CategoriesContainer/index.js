import CategoriesContainer from './CategoriesCotainer';

export default CategoriesContainer;
