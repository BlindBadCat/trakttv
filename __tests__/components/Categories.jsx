import React from 'react';
import { shallow } from 'enzyme';
import CategoriesContainer from '../../src/components/CategoriesContainer/CategoriesCotainer';

describe('<CategoriesContainer />', () => {
  let container;
  const handleSelect = jest.fn();

  beforeEach(() => {
    container = shallow(<CategoriesContainer handleSelect={handleSelect} />);
  });

  it('should render component', () => {
    container.find('select').simulate('change');
    expect(handleSelect.mock.calls.length).toEqual(1);
  });

  it('should has options', () => {
    expect(container.find('option').length).toMatchSnapshot();
  });
});
