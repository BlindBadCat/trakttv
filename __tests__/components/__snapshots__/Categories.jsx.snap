// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoriesContainer /> should has options 1`] = `10`;
